crypto/dilithium/p_pqdsa_test.cc

```diff
@@ -0,0 +1,748 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <gtest/gtest.h>
+#include <openssl/base.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+#include <openssl/mem.h>
+#include <openssl/obj.h>
+#include "../test/test_util.h"
+
+#include <vector>
+#include "../crypto/evp_extra/internal.h"
+#include "../fipsmodule/evp/internal.h"
+#include "../internal.h"
+#include "internal.h"
+
+#ifdef ENABLE_DILITHIUM
+
+#include "../test/file_test.h"
+#include "../test/test_util.h"
+#include "../rand_extra/pq_custom_randombytes.h"
+#include "ml_dsa.h"
+
+// mldsa65kPublicKey is an example ML-DSA-65 public key
```

Comment:
how did you generate it?

Reply:
```
// given an ML-DSA PKEY with a public key present:

// extract public key
std::vector<uint8_t> pub_buf(pk_len);
size_t pub_len;
// get size
EVP_PKEY_get_raw_public_key(pkey.get(), nullptr, &pub_len);
// fill buffer with pk
EVP_PKEY_get_raw_public_key(pkey.get(), pub_buf.data(), &pub_len);

// encode public key as DER
bssl::ScopedCBB cbb;
uint8_t *der;
size_t der_len;
CBB_init(cbb.get(), 0);
EVP_marshal_public_key(cbb.get(), pkey_pk_new.get());
CBB_finish(cbb.get(), &der, &der_len);

```

Reply:
we usually get test examples from "independent" implementations instead of our own. Are you aware of another source you can take the test value from?

Reply:
Will be pulling examples from an independent implementation at https://github.com/lamps-wg/dilithium-certificates/pull/51/files#diff-d2a836a5afde5fbfd8925500dc1e25dfc63f78bc4749ff36c62697e3a0c0355e